### DISCUSSION TITLE
Edgedb-Java #Priorities.

### DISCUSSION THREAD

**Discussion by ShaileshSurya** [geldata/gel#1962] [General]
Here are some of the items that, We should be prioritizing for edged-java driver. 

1. Making Basic API's Work. 
2. Un-marshalling the DB response into Java objects. 
3. Blocking IO
4. Non-Blocking IO.
5. Async Ecosystem. 

We had an initial discussion on this thread https://github.com/edgedb/edgedb/discussions/1923#discussioncomment-116750

Please feel free to suggest otherwise.

**Comment by 1st1** [comment]
I'd be curious to hear your thoughts on 

>Blocking IO
>Non-Blocking IO.
>Async Ecosystem.

action items. AFAIK there multiple different approaches to NIO in Java (and even more so in Scala) with different implementation of Future/Promise objects and different IO primitives. Are there good examples of Java libraries (DB clients?) that support a multitude of different NIO frameworks, different Java languages (Scala, Kotlin) and blocking IO all at the same time?

**Comment by 1st1** [comment]
Also we're making a lot of progress with our Go driver, you can see it here: https://github.com/edgedb/edgedb-go

Some of the discussions and APIs designs are directly applicable to the design of Java APIs.
